src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -880,10 +880,40 @@ void VM_Version::get_processor_features() {
     UseAVX = 0;
   }
 
+  // UseSSE is set to the smaller of what hardware supports and what
+  // the command line requires.  I.e., you cannot set UseSSE to 2 on
+  // older Pentiums which do not support it.
+  int use_sse_limit = 0;
+  if (UseSSE > 0) {
+    if (UseSSE > 3 && supports_sse4_1()) {
+      use_sse_limit = 4;
+    } else if (UseSSE > 2 && supports_sse3()) {
+      use_sse_limit = 3;
+    } else if (UseSSE > 1 && supports_sse2()) {
+      use_sse_limit = 2;
+    } else if (UseSSE > 0 && supports_sse()) {
+      use_sse_limit = 1;
+    } else {
+      use_sse_limit = 0;
+    }
+  }
+  if (FLAG_IS_DEFAULT(UseSSE)) {
+    FLAG_SET_DEFAULT(UseSSE, use_sse_limit);
+  } else if (UseSSE > use_sse_limit) {
+    warning("UseSSE=%d is not supported on this CPU, setting it to UseSSE=%d", (int) UseSSE, use_sse_limit);
+    FLAG_SET_DEFAULT(UseSSE, use_sse_limit);
+  } else if (UseSSE < 0) {
```

Comment:
`UseSSE < 0` check is redundant since the flag is already constrained to `[0..99]` range.

Reply:
Fixed. (Should such ranges be narrowed down to the actual values supported?)

Reply:
> Should such ranges be narrowed down to the actual values supported?

Yes. 

At some point, `UseAVX` was set to `99` by default and then downgraded to the actual version. (BTW the same still happens for `UseSSE`.) I think the highest supported version should be used as the default instead.

Feel free to file a followup RFE or fix it right away.

Reply:
Also, as another cleanup idea, it makes sense to convert `UseSEE`/`UseAVX` from `intx` to `int` and get rid of `(int)` casts.

Reply:
Might need a CSR since we'd upgrade from a warning to an error. I'll defer that to a follow-up.